inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -215,68 +212,115 @@ def arxiv_plot_extract(obj, eng):
-        except KeyError:
+            file_list = untar(tarball_file, scratch_space)
+        except InvalidTarball:
             obj.log.info(
-                'Skipping author list extraction, no tarball with name "%s" found' % filename
+                'Invalid tarball %s for arxiv_id %s',
+                tarball.file.uri,
+                arxiv_id,
             )
             return
 
-        with TemporaryDirectory(prefix='author_list') as scratch_space, \
-                retrieve_uri(tarball.file.uri, outdir=scratch_space) as tarball_file:
-            try:
-                file_list = untar(tarball_file, scratch_space)
-            except InvalidTarball:
-                obj.log.info(
-                    'Invalid tarball %s for arxiv_id %s',
-                    tarball.file.uri,
-                    arxiv_id,
-                )
-                return
+        obj.log.info('Extracted tarball to: {0}'.format(scratch_space))
+        xml_files_list = [path for path in file_list if path.endswith('.xml')]
+        obj.log.info('Found xmlfiles: {0}'.format(xml_files_list))
+
+        extracted_authors = []
+
+        for xml_file in xml_files_list:
+            with open(xml_file, 'r') as xml_file_fd:
+                xml_content = xml_file_fd.read()
+            match = REGEXP_AUTHLIST.findall(xml_content)
+            if match:
+                obj.log.info('Found a match for author extraction')
+
+            extracted_authors.extend(extract_authors_from_xml(xml_content))
+
+        if extracted_authors:
+            for author in extracted_authors:
+                author['full_name'] = decode_latex(author['full_name'])
+
+            obj.data['authors'] = extracted_authors
+
+            set_mark(obj, "authors_xml", True)
+        else:
+            set_mark(obj, "authors_xml", False)
+
+
+def extract_authors_from_xml(xml_content):
+    builder = LiteratureBuilder()
+    if isinstance(xml_content, str):
+        xml_content = xml_content.decode('utf-8')
+
+    # Probably the %auto-ignore comment exists, so we skip the
+    # first line. See: inspirehep/inspire-next/issues/2195
+    if "%auto-ignore" in xml_content:
+        xml_content = xml_content.split('\n', 1)[1]
+
+    content = Selector(text=xml_content, type="xml")
+
+    content.remove_namespaces()
+    # Going through all the authors in the file
+    for author in content.xpath("//Person"):
+
+        ids = []
+        affiliations = []
+        affiliations_identifiers = []
+
+        # Getting all the author ids
+        for source, id in itertools.izip(author.xpath('./authorIDs/authorID[@source!="" and text()!=""]/@source | ./authorids/authorid[@source!="" and text()!=""]/@source').getall(), author.xpath('./authorIDs/authorID[@source!="" and text()!=""]/text() | ./authorids/authorid[@source!="" and text()!=""]/text()').getall()):
+            if not re.match("undefined|inspire-", source, flags=re.IGNORECASE) or re.match("undefined", id, flags=re.IGNORECASE):
```

Comment:
`inspire-` can be present in the ID, not the source.

Reply:
I'm only checking it from the source.

Reply:
exactly, which is not where it can appear in the XML file, it might appear in the ID because it's a prefix of the correct ID, but should not be taken in that case.

Reply:
Ah now I understand. Thank you for explaining.